draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,76 @@ SimulationMarker = {
 
-# Event definition guidelines
+# Event extensibility {#event-extensibility}
+
+The main qlog schema defined by this document describes log and trace formats
+that have related events. A minimal set of common events are presented in
+{{events}}. It is expected that logging is extended with specific, per-protocol
+event definitions that specify the name (category + type) and data needed for
+each individual event. Examples include the QUIC event definitions {{QLOG-QUIC}}
+and HTTP/3 event definitions {{QLOG-H3}}.
+
+New event definitions SHOULD follow the guidance in this section, which provides
+consistency that makes it easier for qlog implementers to support a wide variety
+of protocols.
+
+## Extended Event Schema
+
+New event definitions SHOULD be part of an extension schema, using the
+annotations and concepts presented in this section.
+
+An extension schema defines one or more qlog event categories that can contain
+multiple event definitions. Each category MUST have a globally unique category
+identifier. Each event MUST only belong to a single category. Per
+{{{#name-field}}}, event names are the concatenation of category and type.
+Therefore, these requirements ensure there are no ambiguities when multiple
+extension schema are in use.
+
+Implementations that might record events from extension schemas SHOULD list all
+category identifiers in use. This is achieved by including the appropriate URI
+in the `additional_event_schemas` field of the QlogFile ({{qlog-file-schema}})
+or QlogFileSeq ({{qlog-file-seq-schema}}). The `additional_event_schema` is a
+hint to tools about the possible events that a qlog file might contain. The file
+may contain events that do not belong to a listed category identifier. Tools
+MUST NOT treat this as an error; see {{tooling}}.
+
+Each extension schema is named by a URI. That URI MUST be absolute and MUST
+include a category identifier, indicated using a fragment identifier (characters
+after a "#" in the URI).
+
+For extension schema defined in RFCs, the URI used SHOULD be a URN starting with
+`urn:ietf:params:qlog:` followed by a registered, descriptive name.
+
+Private or non-standard extension schema a can use other URI formats. URIs that
+contain a domain name SHOULD also contain a month-date in the form mmyyyy. The
+definition of the schema and assignment of the URI MUST have been authorized by
+the owner of the domain name on or very close to that date. (This avoids
+problems when domain names change ownership.)
+
+In the following example, a qlog file contains events defined in two hypothetical standardized
+extension schema, along with a private extension. The first schema is named
+"rick" and consists of categories "roll", "astley", and "moranis". The second
+schema is named "john" and consists of categories "doe", "candy", and "goodman".
+The private schema is named "pickle" and consists of categories "pepper",
+"lilly", and "rick":
```

Comment:
As I said above, I think it would REALLY help to have this example (or part of it) at the start of this section, as it makes it much clearer what you mean by schema and category

Reply:
That's specification via example, which is bad. The language needs to be clear enough on its own other wise people only code to the example.

Reply:
I agree that wouldn't be ideal, but then indeed we need to have better text imo :) 

Maybe something more generic up top, like:

``` urn:ietf:params:qlog:<SCHEMA-NAMESPACE>#<CATEGORY-IDENTIFIER> 

...

For example, this document defines the "main" schema namespace with two category identifiers: "generic" and "simulation".

``` 

to help explain how the concepts map to the URIs.